src/ImageSharp/Formats/Tiff/TiffBitsPerSample.cs

```diff
@@ -1,96 +1,242 @@
+        {
+            this.Channel0 = (ushort)Numerics.Clamp(channel0, 1, 32);
+            this.Channel1 = (ushort)Numerics.Clamp(channel1, 0, 32);
+            this.Channel2 = (ushort)Numerics.Clamp(channel2, 0, 32);
+        }
+
+        /// <summary>
+        /// Tries to parse a ushort array and convert it into a TiffBitsPerSample struct.
+        /// </summary>
+        /// <param name="value">The value to parse.</param>
+        /// <param name="sample">The tiff bits per sample.</param>
+        /// <returns>True, if the value could be parsed.</returns>
+        public static bool TryParse(ushort[] value, out TiffBitsPerSample sample)
+        {
+            if (value is null || value.Length == 0)
+            {
+                sample = default;
+                return false;
+            }
+
+            ushort c2;
+            ushort c1;
+            ushort c0;
+            switch (value.Length)
+            {
+                case 3:
+                    c2 = value[2];
+                    c1 = value[1];
+                    c0 = value[0];
+                    break;
+                case 2:
+                    c2 = 0;
+                    c1 = value[1];
+                    c0 = value[0];
+                    break;
+                default:
+                    c2 = 0;
+                    c1 = 0;
+                    c0 = value[0];
+                    break;
+            }
+
+            sample = new TiffBitsPerSample(c0, c1, c2);
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+            => obj is TiffBitsPerSample sample && this.Equals(sample);
+
+        /// <inheritdoc/>
+        public bool Equals(TiffBitsPerSample other)
+            => this.Channel0 == other.Channel0
+               && this.Channel1 == other.Channel1
+               && this.Channel2 == other.Channel2;
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+            => HashCode.Combine(this.Channel0, this.Channel1, this.Channel2);
+
+        /// <summary>
+        /// Converts the bits per sample struct to an ushort array.
+        /// </summary>
+        /// <returns>Bits per sample as ushort array.</returns>
+        public ushort[] ToArray()
+        {
+            if (this.Channel1 == 0)
+            {
+                return new[] { this.Channel0 };
+            }
+
+            if (this.Channel2 == 0)
+            {
+                return new[] { this.Channel0, this.Channel1 };
+            }
+
+            return new[] { this.Channel0, this.Channel1, this.Channel2 };
+        }
+
+        /// <summary>
+        /// Maps an array of bits per sample to a concrete struct value.
+        /// </summary>
+        /// <param name="bitsPerSample">The bits per sample array.</param>
+        /// <returns>TiffBitsPerSample enum value.</returns>
+        public static TiffBitsPerSample? GetBitsPerSample(ushort[] bitsPerSample)
```

Comment:
I don't think we should expose this. Ideally the caller would use the result of `TryParse` within which we can incorporate some of this logic.

Reply:
yeah your right, that method is totally unnecessary now. I overlooked that. Its removed now.